### PR TITLE
Expand Launchable into ubuntu workflow

### PR DESCRIPTION
**Pull request by ono-max**
Follow-up to https://github.com/ruby/ruby/pull/9777

Since the integration of Launchable into the macOS workflow has been completed in the previous PR, this PR extends the integration to the Ubuntu workflow as well. Along with this integration, I've also refactored some logic.

### CODE REVIEW

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -13,6 +13,21 @@ on:
...
+  # https://github.com/launchableinc/cli/blob/v1.80.1/launchable/utils/authentication.py#L20
+  LAUNCHABLE_ORGANIZATION: ${{ github.repository_owner }}
+  LAUNCHABLE_WORKSPACE: ${{ github.event.repository.name }}
+  # https://github.com/launchableinc/cli/blob/v1.80.1/launchable/utils/authentication.py#L71
+  GITHUB_PR_HEAD_SHA: ${{ github.event.pull_request.head.sha || github.sha }}
+  # This secret setting is needed if you want to run Launchable on your forked
+  # repository.
+  # See https://github.com/ruby/ruby/wiki/CI-Servers#launchable-ci for details.
+  LAUNCHABLE_TOKEN: ${{ secrets.LAUNCHABLE_TOKEN }}
+
```
[the remark this answers is not in the document]
Comment (ono-max) [reply .github/workflows/ubuntu.yml]: Updated.

File: .github/workflows/ubuntu.yml [hunk 2]
```diff
@@ -54,6 +70,17 @@ jobs:
...
+        id: enable_launchable
+        run: echo "enable_launchable=true" >> $GITHUB_OUTPUT
+        working-directory:
+        if: >-
+          ${{
+          (github.repository == 'ruby/ruby' ||
+          (github.repository != 'ruby/ruby' && env.LAUNCHABLE_TOKEN)) &&
+          (matrix.test_task == 'check' || matrix.test_task == 'test-all')
+          }}
+
```
[the remark this answers is not in the document]
Comment (ono-max) [reply .github/workflows/ubuntu.yml]: Updated.

File: .github/workflows/ubuntu.yml [hunk 3]
```diff
@@ -115,6 +146,15 @@ jobs:
...
 
+      - name: Record test results in Launchable
+        uses: ./.github/actions/setup/launchable/record-test
+        with:
+          # We need to configure the `build` directory because
+          # this composite action is executed in the default working directory.
+          report-path: build/launchable_reports.json
+          os: ubuntu-20.04
+        if: ${{ always() && steps.enable_launchable.outputs.enable_launchable }}
+
```
[the remark this answers is not in the document]
Comment (ono-max) [reply .github/workflows/ubuntu.yml:131]: Thank you for letting me know 👍

File: .github/workflows/ubuntu.yml [hunk 4]
```diff
@@ -25,6 +40,7 @@ jobs:
     strategy:
       matrix:
         test_task: [check]
+        test_opts: ['']
```
[the remark this answers is not in the document]
Comment (ono-max) [reply .github/workflows/ubuntu.yml:28]: You're right. Thank you. Let me fix it.

**Comment by hsbt** [comment]
Nitpick. 

`.github/actions/setup/launchable/setup/action.yml` is redundant name because `setup` is used twice. Should we use `.github/actions/launchable/setup/action.yml` and `.github/actions/launchable/record-test/action.yml` ?

**Comment by ono-max** [comment]
@hsbt 

Thank you for your suggestion. Updated 👍

**Review by hsbt — APPROVED** [review]

**Pull request merged**